riscv/triggers.cc

```diff
@@ -234,6 +234,62 @@ std::optional<match_result_t> mcontrol_common_t::detect_memory_access_match(proc
   return std::nullopt;
 }
 
+reg_t mcontrol6_t::tdata1_read(const processor_t * const proc) const noexcept {
+  unsigned xlen = proc->get_const_xlen();
+  reg_t tdata1 = 0;
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_TYPE(xlen), 6);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_DMODE(xlen), dmode);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_VS, proc->extension_enabled('H') ? vs : 0);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_VU, proc->extension_enabled('H') ? vu : 0);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_HIT, hit);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_SELECT, select);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_TIMING, timing);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_ACTION, action);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_CHAIN, chain);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_MATCH, match);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_M, m);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_S, s);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_U, u);
```

Comment:
Don't these need to consider `proc->extension_enabled_const('S')`, like `mcontrol_t` does?

Reply:
We check the ```proc->extension_enabled_const('S')``` in tdata1_write() but not in tdata1_read().